lib/parser.js

```diff
@@ -341,7 +341,15 @@ module.exports = function (args, opts, y18n) {
       var configPath = argv[configKey] || configLookup[configKey]
       if (configPath) {
         try {
-          var config = require(path.resolve(process.cwd(), configPath))
+          var config = null
+          var resolvedConfigPath = path.resolve(process.cwd(), configPath)
+
+          if (typeof flags.configs[configKey] === 'function') {
+            config = flags.configs[configKey](resolvedConfigPath)
+            if (config instanceof Error) error = config
```

Comment:
If `config` is an `Error`, then the enumerable properties from the error will be applied to argv in this function below, which, I'm assuming, is not intended.

Perhaps throw the error here instead, or maybe just return after setting `error`?